column_buffer.go

```diff
@@ -617,49 +617,40 @@ func (col *repeatedColumnBuffer) Swap(i, j int) {
 }
 
 func (col *repeatedColumnBuffer) WriteValues(values []Value) (numValues int, err error) {
-	// The values may belong to the last row that was written if they do not
-	// start with a repetition level less than the column's maximum.
-	var continuation Row
-	if len(values) > 0 && values[0].repetitionLevel != 0 {
-		continuation, values = splitRowValues(values)
-	}
-
-	if len(continuation) > 0 {
-		for i, v := range continuation {
-			if v.definitionLevel == col.maxDefinitionLevel {
-				if _, err := col.base.WriteValues(continuation[i : i+1]); err != nil {
-					return numValues, err
-				}
-			}
-			col.repetitionLevels = append(col.repetitionLevels, v.repetitionLevel)
-			col.definitionLevels = append(col.definitionLevels, v.definitionLevel)
-			numValues++
-		}
-	}
-
-	maxNumValues := 0
+	maxRowLen := 0
 	defer func() {
-		clearValues(col.buffer[:maxNumValues])
+		clearValues(col.buffer[:maxRowLen])
 	}()
 
-	var row []Value
+	for i := 0; i < len(values); {
+		j := i
+
+		if values[j].repetitionLevel == 0 {
+			j++
+		}
+
+		for j < len(values) && values[j].repetitionLevel != 0 {
+			j++
+		}
 
-	for len(values) > 0 {
-		row, values = splitRowValues(values)
-		if err := col.writeRow(row); err != nil {
+		if err := col.writeRow(values[i:j]); err != nil {
 			return numValues, err
 		}
-		numValues += len(row)
-		if len(col.buffer) > maxNumValues {
-			maxNumValues = len(col.buffer)
+
+		if len(col.buffer) > maxRowLen {
+			maxRowLen = len(col.buffer)
 		}
+
+		numValues += j - i
+		i = j
```

Comment:
There were issues with the way continuations of rows were handled (e.g. write values of a row to a column buffer in multiple calls to `WriteValues`).